inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_request_internal.hpp

```diff
@@ -109,7 +109,9 @@ class InferRequestInternal : virtual public IInferRequestInternal {
             if (preProcRequired) {
                 addInputPreProcessingFor(name, userBlob, devBlob ? devBlob : _inputs[name]);
             } else {
-                size_t inputSize = details::product(foundInput->getTensorDesc().getDims());
+                size_t inputSize = foundInput->getTensorDesc().getLayout() != InferenceEngine::Layout::SCALAR
+                ? InferenceEngine::details::product(foundInput->getTensorDesc().getDims())
```

Comment:
I suppose the same should be done for `outputSize` below on lone 127?

Reply:
@adk9 Could you fix this comment?

Reply:
I noticed that a follow-up PR (#3821) was created. Thanks, @ilya-lavrenov!